src/ShopCertification.php

```diff
@@ -117,11 +118,20 @@ public function setOrderId($orderId)
     /**
      * @param string $productItemId Must be same as ITEM_ID provided in the Heureka XML feed.
      *
-     * @return self
+     * @return ShopCertification
+     * @throws DuplicateProductItemIdException
      */
     public function addProductItemId($productItemId)
     {
-        $this->productItemIds[] = (string)$productItemId;
+        $productItemId = (string)$productItemId;
+
+        if (isset($this->productItemIds[$productItemId])) {
```

Comment:
to je otazka no, ja myslim ze nejakej student-shop to klidne muze delat a
pak bude volat na podporu :D co @howkey666 ?

2016-04-01 15:33 GMT+02:00 Jakub Chábek <notifications@github.com>:

> In src/ShopCertification.php
> <https://github.com/heureka/overeno-zakazniky/pull/10#discussion_r58205559>
> :
>
> >       */
> >      public function addProductItemId($productItemId)
> >      {
> > -        $this->productItemIds[] = (string)$productItemId;
> > +        $productItemId = (string)$productItemId;
> > +
> > +        if (isset($this->productItemIds[$productItemId])) {
>
> Neviděl jsem e-shop, který by měl nakupování více kusů řešeno přes
> duplikací položek v košíku. Fakt si myslíte, že to může nastat? Protože mě
> se docela tohle řešení líbí (když to někdo posere, tak to uvidí ...)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/heureka/overeno-zakazniky/pull/10/files/cf833dfeb1d854390a7a937fda2552f7054947e5..1e993f39c23c851e5274bc456d25728cfb3ccca2#r58205559>
>